src/test/fuzz/merkleblock.cpp

```diff
@@ -16,12 +16,49 @@
 void test_one_input(const std::vector<uint8_t>& buffer)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
-    std::optional<CPartialMerkleTree> partial_merkle_tree = ConsumeDeserializable<CPartialMerkleTree>(fuzzed_data_provider);
-    if (!partial_merkle_tree) {
-        return;
+
+    {
+        std::optional<CPartialMerkleTree> partial_merkle_tree = ConsumeDeserializable<CPartialMerkleTree>(fuzzed_data_provider);
+        if (partial_merkle_tree) {
+            (void)partial_merkle_tree->GetNumTransactions();
+            std::vector<uint256> matches;
+            std::vector<unsigned int> indices;
+            while (fuzzed_data_provider.ConsumeBool()) {
+                indices.push_back(fuzzed_data_provider.ConsumeIntegral<unsigned int>());
+            }
+            (void)partial_merkle_tree->ExtractMatches(matches, indices);
+        }
+    }
+
+    {
+        CMerkleBlock merkle_block;
+        const std::optional<CBlock> opt_block = ConsumeDeserializable<CBlock>(fuzzed_data_provider);
```

Comment:
It doesn't look like there is any state shared between the two fuzz targets in this file. Often the fuzz engine can only generate random noise and I write initial seeds myself. With a format that has several unrelated seeds concatenated together, this becomes harder.

I guess the options are:
* Split into two fuzz scripts (requires one more binary)
* Use a switch-case to select a fuzz target within the script (requires one more byte at the end of each fuzz input)

Reply:
Thanks for reviewing. Now skipping all changes that are unrelated to CAddrMan to keep this PR focused.